tasks/verify_password.yml

```diff
@@ -0,0 +1,36 @@
+- name: Ensure that the mssql-server service is started
+  service:
+    name: mssql-server
+    state: started
+
+- name: Check if a custom tcpport setting exist
+  command: /opt/mssql/bin/mssql-conf get network tcpport
+  ignore_errors: true
+  changed_when: false
+  register: __mssql_custom_tcp_port
+
+- name: Check if a custom ipaddress setting exist
+  command: /opt/mssql/bin/mssql-conf get network ipaddress
+  ignore_errors: true
+  changed_when: false
+  register: __mssql_custom_ip_address
+
+- name: Check if the set password matches the existing password
+  vars:
+    __ipaddress: >-
+      {{ __mssql_custom_ip_address.stdout | regex_search('[0-9].*') }}
```

Comment:
this regex will match `0abc` or `0` or `0       ;` - do you mean `[0-9.]+` - that is - match a string that looks like an ipv4 address?  That would be `regex_search('[0-9]{1-3}[.][0-9]{1-3}[.][0-9]{1-3}[.][0-9]{1-3}')`
can `network.ipaddress` be an ipv6 address?

I think you would be better off by stripping out the `ipaddress : ` from the string instead
`regex_replace('^.*ipaddress +: +', '')` and just assume that whatever value is left over is the ip address

Reply:
Further tasks in playbook expect __ipaddress to be either an ip address or an empty string. So, instead of stripping `ipaddress : ` I returned the regexes.